pyrate/core/dem_error.py

```diff
@@ -0,0 +1,307 @@
+            azimuth_angle[ix, iy] = az12
+    np.reshape(azimuth_angle, lat.shape)
+
+    # old code: spherical azimuth angle calculation
+    #azimuth_angle2 = np.arccos(np.divide(np.multiply(np.sin(sat_lat), np.cos(lat)) - \
+    #                                np.multiply(np.multiply(np.cos(sat_lat), np.sin(lat)), np.cos(sat_lon - lon)), \
+    #                                np.sin(epsilon)))
+    #azimuth_angle_diff = azimuth_angle2 - azimuth_angle
+    #print(np.nanmin(azimuth_angle_diff), np.nanmax(azimuth_angle_diff))
+    # the difference between Vincety's azimuth calculation and the spherical approximation is ~0.001 radians
+
+    # for validation with GAMMA output only
+    azimuth_angle_gamma = -(azimuth_angle - np.pi / 2) # local heading towards satellite
+    # maximum differences to the GAMMA-derived local azimuth angles for Sentinel-1 test data are within +/-0.5 deg
+    # this could be improved by using orbital state vectors to calculate that satellite positions (see above comment)
+
+    # save angles to npy arrays
+    look_angle_file = os.path.join(params[cf.OUT_DIR],'look_angle.npy')
+    np.save(file=look_angle_file, arr=look_angle)
+
+    # save angles to geotiff
+    gt, md, wkt = shared.get_geotiff_header_info(ifg_path)
+    md[ifc.EPOCH_DATE] = None # needs to have a value in write_output_geotiff
+    look_angle_file = os.path.join(params[cf.OUT_DIR], 'look_angle.tif')
+    shared.write_output_geotiff(md, gt, wkt, look_angle, look_angle_file, np.nan)
+    incidence_angle_file = os.path.join(params[cf.OUT_DIR], 'incidence_angle.tif')
+    shared.write_output_geotiff(md, gt, wkt, incidence_angle, incidence_angle_file, np.nan)
+    azimuth_angle_file = os.path.join(params[cf.OUT_DIR], 'azimuth_angle.tif')
+    shared.write_output_geotiff(md, gt, wkt, azimuth_angle, azimuth_angle_file, np.nan)
+    incidence_angle_gamma_file = os.path.join(params[cf.OUT_DIR], 'incidence_angle_gamma.tif')
+    shared.write_output_geotiff(md, gt, wkt, incidence_angle_gamma, incidence_angle_gamma_file, np.nan)
+    azimuth_angle_gamma_file = os.path.join(params[cf.OUT_DIR], 'azimuth_angle_gamma.tif')
+    shared.write_output_geotiff(md, gt, wkt, azimuth_angle_gamma, azimuth_angle_gamma_file, np.nan)
+
+    return look_angle
+
+
+def calc_local_baseline(ifg, az, look_angle, params):
+    """
+    Function to calculate perpendicular baseline values for each pixel.
+    """
+    # open ifg object
+    if not ifg.is_open:
+        ifg.open()
+
+    # read relevant metadata from IFG
+    prf = float(ifg.meta_data[ifc.PYRATE_PRF_HERTZ])
+    az_looks = int(ifg.meta_data[ifc.PYRATE_AZIMUTH_LOOKS])
+    az_n = int(ifg.meta_data[ifc.PYRATE_AZIMUTH_N])
+    #base_T = float(ifg.meta_data[ifc.PYRATE_BASELINE_T])
+    base_C = float(ifg.meta_data[ifc.PYRATE_BASELINE_C])
+    base_N = float(ifg.meta_data[ifc.PYRATE_BASELINE_N])
+    #baserate_T = float(ifg.meta_data[ifc.PYRATE_BASELINE_RATE_T])
```

Comment:
delete T components from here if they'll never be used

Reply:
Done.